src/server/index.js

```diff
@@ -70,21 +70,25 @@ const server = new ApolloServer({
   resolvers,
   introspection: true,
   plugins: [ApolloServerPluginDrainHttpServer({ httpServer })],
-  formatError: error => {
+  formatError: (formattedError, error) => {
+    log.error({
+      // TODO: request is no longer available in formatError, figure out
+      // another way to do this.
+      // userId: request.user && request.user.id,
+      code: error?.extensions?.code ?? 'INTERNAL_SERVER_ERROR',
+      error: formattedError,
+      msg: "GraphQL error"
+    });
+
     if (process.env.SHOW_SERVER_ERROR || process.env.DEBUG) {
-      if (error instanceof GraphQLError) {
-        return error;
-      }
-      return new GraphQLError(error.message);
+      return formattedError;
     }
 
-    return new GraphQLError(
-      error &&
-      error.originalError &&
-      error.originalError.code === "UNAUTHORIZED"
-        ? "UNAUTHORIZED"
-        : "Internal server error"
-    );
+    // Strip out stacktrace and other potentially sensative details.
+    return {
+      message: formattedError.message,
```

Comment:
Definitely want to find the right balance. There's just a lot of places where giving users errors like "That invitation is no longer valid" or "Not allowed to add contacts after the campaign starts" is a lot more helpful than "Internal server error".

Another possible option, which I thought about but didn't test, would be to create our own error class that extends `GraphQLError` and throw that instead in the places where we want the message to be shown. Then we could check for that type in `formatError`.

Not 100% sure if that's possible, but I think it might be.